src/main/java/frc/team2412/robot/subsystem/ShooterSubsystem.java

```diff
@@ -362,23 +373,62 @@ public void resetHoodEncoder(boolean reset) {
      *            The angle (in degrees) to set the turret to (negative for counterclockwise).
      */
     public void setTurretAngle(double angle) {
+        if (isTurretAtAngle(angle)) {
+            return;
+        }
         if (angle < MIN_TURRET_ANGLE) {
-            System.out.println("LOG: Desired turret angle is below min angle!");
-            if (angle + 360 < MAX_TURRET_ANGLE) {
-                System.out.println("LOG: Targeting desired turret angle in other direction...");
-                angle += 360;
-            } else {
-                System.out.println("LOG: Couldn't wrap around turret angle!");
-            }
+            System.out.println("LOG: Desired turret angle is below min angle, going to highest reachable angle");
+            // angle = Math.min(angle + 360, MAX_TURRET_ANGLE);
+            // angle = MAX_TURRET_ANGLE;
+            loopToMax = true;
+            t.start();
+            // if (angle + 360 < MAX_TURRET_ANGLE) {
+            // System.out.println("LOG: Targeting desired turret angle in other direction...");
+            // angle += 360;
+            // } else {
+            // System.out.println("LOG: Couldn't wrap around turret angle!");
+            // angle = MAX_TURRET_ANGLE;
+            // }
         } else if (angle > MAX_TURRET_ANGLE) {
-            System.out.println("LOG: Desired turret angle is above max angle!");
-            if (angle - 360 > MIN_TURRET_ANGLE) {
-                System.out.println("LOG: Targeting desired turret angle in other direction...");
-                angle -= 360;
-            } else {
-                System.out.println("LOG: Couldn't wrap around turret angle!");
+            System.out.println("LOG: Desired turret angle is above max angle, going to lowest reachable angle");
+            // angle = Math.max(angle - 360, MIN_TURRET_ANGLE);
+            // angle = MIN_TURRET_ANGLE;
+            loopToMin = true;
+            t.start();
+            // if (angle - 360 > MIN_TURRET_ANGLE) {
+            // System.out.println("LOG: Targeting desired turret angle in other direction...");
+            // angle -= 360;
+            // } else {
+            // System.out.println("LOG: Couldn't wrap around turret angle!");
+            // angle = MIN_TURRET_ANGLE;
+            // }
+        }
+
+        if (loopToMax) {
+            angle = MAX_TURRET_ANGLE;
+            if (t.get() > 1 && isTurretAtAngle(MAX_TURRET_ANGLE)) {
+                loopToMax = false;
+                t.reset();
+                t.stop();
```

Comment:
Need to swap these, as t.stop() will simply set the timer to whatever time elapsed between these two lines of code, which may not be zero.  Ditto for below.

Reply:
intended feature see lines above